extension/dnscrypt/dnscrypt.js

```diff
@@ -51,23 +53,38 @@ class DNSCrypt {
     return `127.0.0.1#${this.config.localPort || 8854}`;
   }
 
-  async prepareConfig(config = {}) {
+  async prepareConfig(config = {}, reCheckConfig = false) {
     this.config = config;
     let content = await fs.readFileAsync(templatePath, {encoding: 'utf8'});
     content = content.replace("%DNSCRYPT_FALLBACK_DNS%", config.fallbackDNS || "1.1.1.1");
     content = content.replace("%DNSCRYPT_LOCAL_PORT%", config.localPort || 8854);
     content = content.replace("%DNSCRYPT_LOCAL_PORT%", config.localPort || 8854);
     content = content.replace("%DNSCRYPT_IPV6%", "false");
 
-    let serverList = await this.getServers();
-
-    content = content.replace("%DNSCRYPT_SERVER_LIST%", JSON.stringify(serverList));
-
     const allServers = await this.getAllServers();
 
     content = content.replace("%DNSCRYPT_ALL_SERVER_LIST%", this.allServersToToml(allServers));
 
+    let serverList = await this.getServers();
+    serverList = serverList.filter((serverName) => {
+      if (!serverName) {
+        return false;
+      }
+
+      return (allServers.filter((server) => (server && server.name && server.stamp && server.name === serverName)).length > 0) ? true : false;
```

Comment:
try to avoid code like this which is hard to understand the purpose.

what i wrote:

```javascript
    const allServers = await this.getAllServers();
    const allServerNames = await this.getAllServerNames();
    serverList = serverList.filter((n) => allServerNames.includes(n));
```